src/autopas/cells/ParticleCell.h

```diff
@@ -128,9 +142,50 @@ class ParticleCell {
   [[nodiscard]] virtual std::array<double, 3> getCellLength() const = 0;
 
   /**
-   * Lock object for exclusive access to this cell.
+   * Get the type of particles contained in this cell. Possible values:
+   * dummy: this cell is empty
+   * owned: this cell can ONLY contain owned particles
+   * halo: this cell can ONLY contain halo particles
+   * ownedOrHalo: this cell can contain owned or halo particles
+   * @return type of particles inside this cell
+   */
+  const OwnershipState getPossibleParticleOwnerships() { return _ownershipState; }
+
+  /**
+   * Set the type of particles contained in this cell. Possible values:
+   * dummy: this cell is empty
+   * owned: this cell can ONLY contain owned particles
+   * halo: this cell can ONLY contain halo particles
+   * ownedOrHalo: this cell can contain owned or halo particles
+   * @note: At the moment an ownership of a cell can only be set once.
+   * @param state type of particles inside this cell
+   */
+  void setPossibleParticleOwnerships(OwnershipState state) {
+    std::lock_guard<AutoPasLock> guard(this->_cellLock);
+    if (_ownershipStateDefined) {
+      autopas::utils::ExceptionHandler::exception(
+          "ParticleCell::setPossibleParticleOwnerships() can not set OwnershipState of a cell after "
+          "it has been set");
+    }
+    _ownershipState = state;
+    _ownershipStateDefined = true;
+  }
+
+  /**
+   * Lock object for exclusive access to this cell. This lock has to be mutable since it is set in
+   * getNumberOfParticles() const.
+   */
+  mutable AutoPasLock _cellLock{};
+
+ protected:
```

Comment:
```suggestion
 protected:
  /**
   * Lock the object for exclusive access to this cell. This lock has to be mutable since it is set in
   * getNumberOfParticles() const.
   */
  mutable AutoPasLock _cellLock{};
```
Why was this made public?

Reply:
As far as I can see, this lock is only used from outside in `checkParticleInCellAndUpdateByIDAandPosition()` from `ParticleCellHelpers.h`. But I could write a getter for this lock, then we can move it to protected section?

Reply:
yes a getter is fine, just no setter ;)

Reply:
done in [94c85bf](https://github.com/AutoPas/AutoPas/pull/782/commits/94c85bf7a061dbad59c9a4e2f931bbd52c27b653)